### PR TITLE
Bridge Validator App Scaffold

### PR DESCRIPTION
**Pull request by jannikluhn**
Does nothing but read and validate a config file.

### CODE REVIEW

File: bridge/bridge/main.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+)
+def main(config_path):
+    try:
+        config = load_config(config_path)
+    except toml.decoder.TomlDecodeError as decode_error:
+        raise click.UsageError(
+            f"Error in config file: {decode_error}"
+        ) from decode_error
+    except ValueError as value_error:
+        raise click.UsageError(f"Error in config file: {value_error}") from value_error
```
[the remark this answers is not in the document]
Comment (jannikluhn) [reply bridge/bridge/main.py]: Changed it to just "invalid config" as click will prepend `"Error"` and the exception will provide details.

**Comment by jannikluhn** [comment]
> Therefore I would argue to at least check the types of the user configuration. A named tuple could help here.

Type hints don't do anything at runtime, so a named tuple wouldn't help. I've added individual validation functions though. In addition, I played around a little with `TypedDict` but couldn't come up with anything that added any value, so I kept the plain dict. Open for suggestions though (which we could also implement in a future PR).

> Moreover I think using abbreviations within the code is fine if everyone is aware about it or it is just common to use. But on user interactions/output I would always prefer to write them out. I refer here to config -> configuration. Ignore it if you don't feel the same.

Agree in general, but IMO `config` is common enough to use it.

**Comment by jannikluhn** [comment]
Oh, and about RPC URL validation: I think there's quite a lot of different formats and I'd rather not try to validate this. If we want it and it's possible, we should open a PR to `web3.py`. I think the proper place to do it would be the instantiation of the `HTTPProvider` (so not only at the first request).

File: bridge/bridge/main.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+@click.command()
+@click.option(
+    "-c", "--config", "config_path", type=click.Path(exists=True), default=None
+)
+def main(config_path):
+    try:
+        config = load_config(config_path)
+    except toml.decoder.TomlDecodeError as decode_error:
+        raise click.UsageError(
+            f"Error in config file: {decode_error}"
```
[the remark this answers is not in the document]
Comment (weilbith) [reply bridge/bridge/main.py]: Handled

**Comment by jannikluhn** [comment]
Added a Makefile and hvae mypy and pytest run in CI. I've mostly copy and pasted and have no idea what I actually did, but it seems to work. Please have another look

**Comment by weilbith** [comment]
> I think there's quite a lot of different formats and I'd rather not try to validate this.

Probably [try to parse the url](https://docs.python.org/3/library/urllib.parse.html) and validate the protocol is a simple alternative?

**Comment by jannikluhn** [comment]
> Probably try to parse the url and validate the protocol is a simple alternative?

I don't see a good reason for limiting the protocols one may use, but I'd be up for the `urlparse` approach. However, I played around with it a little but couldn't produce an invalid URL. Is there actually one? If not validating wouldn't make sense again.

**Comment by schmir-at-bb** [comment]
We're not in the business of writing validating config file parsers. I think the configuration file validation is good enough already.

**Pull request merged**